### DISCUSSION TITLE
With Google set as Auth provider, users are not saved inside the users collection. Is this the expected behavior?

### DISCUSSION THREAD

**Discussion by amanzan** [pocketbase/pocketbase#1120] [Q&A]
Hi! I'm very new playing around with PocketBase, so please bear with me if I ask something too obvious.

I obtained a Client Id and Client Secret from my Google account, registered the redirect URL, and then enabled and configured Google as the Auth provider in my PocketBase account.

I'm able to sign in, I get an access token and everything seems correct.

Is it expected that nothing is saved in the Users collection in PocketBase? If it's expected, what would I need to do to save basic user info for every new user that signs in?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the "I'm able to sign in, I get an access token and everything seems correct." part. How are you obtaining an access token? Usually you'll get only an access code and then send a request to the PocketBase backend to exchange the code for a token.

For a full web JS example integration you could check https://pocketbase.io/docs/authentication/#web-oauth2-integration

**Comment by amanzan** [comment]
Sorry, I should've been more specific. I'm signing in from an Android device, using an oauth2 library, so that's how I get the access token. Then I'll be getting data from another collection by making API calls and adding this access token as a header. 
I was just wondering if a list of users could be saved, to keep track of basic info of all users that will open the app and sign in.

**Reply by ganigeorgiev** [reply]
>  I'm signing in from an Android device, using an oauth2 library, so that's how I get the access token. 

No, this will not work.
You'll have to first open the user browser (or similar) to authenticate with Google and then via deep link to catch the redirect url and send a `POST 
/api/collections/users/auth-with-oauth2` request with the **access code** from the deeplink in order to allow PocketBase to create/link your user.

You could also check this discussion https://github.com/pocketbase/pocketbase/discussions/442 (it is for Flutter but @pchasco noted some Android specifics in  https://github.com/pocketbase/pocketbase/discussions/442#discussioncomment-4160820 that you may find useful).

**Reply by ganigeorgiev** [reply]
_The API docs for the above call could be found either under the "API Preview" section for the `users` collection in the Admin UI or in https://pocketbase.io/docs/api-records/#auth-record-actions._

**Comment by pchasco** [comment]
I can push up a repo with this working for Google auth using flutter and
flutter_web_auth later today. Not near a computer right now.
